domain-management/src/main/java/org/jboss/as/domain/management/security/WrapperSSLContext.java

```diff
@@ -122,8 +122,7 @@ private WrapperSSLSocketFactory(final SSLSocketFactory wrapped) {
 
             @Override
             public Socket createSocket(Socket s, String host, int port, boolean autoClose) throws IOException {
-                // TODO Auto-generated method stub
-                return null;
+                return wrapped.createSocket(s, host, port, autoClose);
```

Comment:
Before returning it, should pass the created socket to the setSslParams(Socket) method, like the other variants do.

While looking at this, I (aka my IDE) noticed that this variant recurses infinitely:

{code}
            public Socket createSocket(InetAddress address, int port, InetAddress localAddress, int localPort)
                    throws IOException {
                Socket socket = createSocket(address, port, localAddress, localPort);
                setSslParams(socket);
                return socket;
            }
{code}

That call createSocket call should be changed to wrapped.createSocket(...)

Do you mind fixing that in this pull request?